include/aws/common/byte_buf.h

```diff
@@ -348,6 +348,16 @@ int aws_byte_buf_append_dynamic(struct aws_byte_buf *to, const struct aws_byte_c
 AWS_COMMON_API
 int aws_byte_buf_reserve(struct aws_byte_buf *buffer, size_t requested_capacity);
 
+/**
+ * Convenience function that attempts to increase the capacity of a buffer relative to the current
+ * length.
+ *
+ *  aws_byte_buf_reserve_from_len(buf, x) ~~ aws_byte_buf_reserve(buf, buf->len + x)
+ *
+ */
+AWS_COMMON_API
+int aws_byte_buf_reserve_from_len(struct aws_byte_buf *buffer, size_t additional_length);
```

Comment:
[nitpick] Feels like _grow or _expand or _extend, do we not have this pattern anywhere else? reserve_from_len just feels a bit unintuitive to me offhand.

Reply:
_reserve_relative was another possibility